### PR TITLE
Add support for migrating from SmartMatrix's bottom to top stacking.

### PR DESCRIPTION
**Pull request by echamberlain**
Added a pixel mapper to accommodate migration from SmartMatrix library (see https://github.com/pixelmatix/SmartMatrix) to rpi-rgb-led-matrix.

SmartMatrix has a configuration option (SMARTMATRIX_OPTIONS_BOTTOM_TO_TOP_STACKING) for bottom to top stacking of panels.  This pull request adds that functionality.

I needed this modification to migrate my 64x64 matrix from SmartMatrix/Teensy to a Raspberry Pi with the Adafruit RGB matrix HAT without having to take my display apart and reorient the panels.

This code is untested with parallel chains.

Usage example:
`sudo ./demo -D3 --led-chain=4 --led-pixel-mapper=BottomToTop-mapper`

img_0484
img_0483

### CODE REVIEW

**Comment by hzeller** [comment]
Cool!
Can you also add a description of this mapper in the examples-api-use/README.md file ?

**Comment by echamberlain** [comment]
Sure.

**Comment by hzeller** [comment]
Cool. One more thing: can you match the C++ indentation style (2 spaces) with the rest of the file ?

File: examples-api-use/README.md [hunk 1]
```diff
@@ -270,6 +270,42 @@ as parameter after a colon:
...
+This mapper is especially useful if you are migrating from the 
+[SmartMatrix](https://github.com/pixelmatix/SmartMatrix) library and don't want to
+have to reorient your panels.  SmartMatrix refers to this configuration
+as SMARTMATRIX_OPTIONS_BOTTOM_TO_TOP_STACKING.
+
+```
+So the following chain
+    [<][<][<][<] }- Raspbery Pi connector
+
+is arranged in this U-shape (on its side)
```
Comment (hzeller) [inline examples-api-use/README.md]: This is probably meant to be S-shape

**Comment by mbates14** [comment]
Ooooo this could do exactly what I need. How do I get this mapper?

**Comment by hzeller** [comment]
You can download the patch via https://github.com/hzeller/rpi-rgb-led-matrix/pull/612.patch @mbates14 
Then go in the toplevel directory and patch it with the `patch` utility.

```
cd rpi-rgb-led-matrix
wget https://github.com/hzeller/rpi-rgb-led-matrix/pull/612.patch
patch -p1 < 612.patch
```

This pull request is not merged yet into the main distribution as there are some pending change requests.

**Comment by echamberlain** [comment]
Updated the indentation.  Sorry for the delay.

**Comment by mbates14** [comment]
Well, I added it according to your steps but it didnt work. So I typed make again, and then it worked with the Demo C files. But it is showing no such mapper with my python files. So i tried make python again and still nothing. Ugh. Sometimes I wish documentation was better. Or maybe I was better at it...

**Comment by hzeller** [comment]
This should essentially the same as the V-Mapper by @marcmerlin

**Comment by marcmerlin** [comment]
Oh no, you mostly wrote the same coe than me and it never got merged all this time. Sorry 
@echamberlain /  @mbates14 that I didn't find it and ended up rewriting a similar one (it also allows flipping the panels upside down to use shorter cables, see pictures here http://marc.merlins.org/perso/arduino/post_2020-03-13_RGB-Panels_-from-192x80_-to-384x192_-to-384x256-and-maybe-not-much-beyond.html )

**Pull request closed without merging**